stdlib/universe/fill.go

```diff
@@ -223,13 +220,22 @@ func (t *fillTransformation) Process(id execute.DatasetID, tbl flux.Table) error
 
 	var fillValue interface{}
 	if !t.spec.UsePrevious {
-		if tbl.Cols()[colIdx].Type != flux.ColumnType(t.spec.Value.Type()) {
+		if colIdx > -1 && tbl.Cols()[colIdx].Type != flux.ColumnType(t.spec.Value.Type()) {
 			return errors.Newf(codes.FailedPrecondition, "fill column type mismatch: %s/%s", tbl.Cols()[colIdx].Type.String(), flux.ColumnType(t.spec.Value.Type()).String())
 		}
 		fillValue = values.Unwrap(t.spec.Value)
 	}
 
-	table, err := table.StreamWithContext(t.ctx, key, tbl.Cols(), func(ctx context.Context, w *table.StreamWriter) error {
+	tableCols := tbl.Cols()
+	if colIdx < 0 {
+		c := flux.ColMeta{
+			Label: t.spec.Column,
+			Type:  flux.ColumnType(t.spec.Value.Type()),
+		}
+		tableCols = append(tableCols, c)
```

Comment:
This can potentially cause a race condition. Imagine a situation where the original columns is defined in this way:

```
cols := make([]flux.ColMeta, 0, 1)
cols = append(cols, flux.ColMeta{Label: "mycol", Type: flux.TString})
```

If we call `append()` later on these columns, the `append()` function will notice that the capacity would be exceeded and it will allocate a new slice with the appropriate capacity. This works perfectly and is the expected pathway for this code.

On the other hand, if we did this:

```
cols := make([]flux.ColMeta, 0, 2)
cols = append(cols, flux.ColMeta{Label: "mycol", Type: flux.TString})
```

This would also look like it works fine at the allocations site. The length is 1 and the capacity is 2. We overallocated but there's no problem with the slice itself.

When we retrieve the slice here and use `append()`, we put the new column inside of the additional section of capacity. If we did this in two different locations, then we would be modifying the same buffer and it would open ourselves up to a race condition.

Instead, we should do this:

```
newCols := make([]flux.ColMeta, len(tableCols), len(tableCols)+1)
copy(newCols, tableCols)
tableCols = append(newCols, c)
```